src/RandomEnchants.cpp

```diff
@@ -1,63 +1,139 @@
-/*
-* Converted from the original LUA script to a module for Azerothcore(Sunwell) :D
-*/
 #include "ScriptMgr.h"
-#include "Player.h"
 #include "Configuration/Config.h"
+#include "ObjectMgr.h"
 #include "Chat.h"
+#include "Player.h"
+#include "Object.h"
 
-class RandomEnchantsPlayer : public PlayerScript{
+bool RandomEnchantEnabled = true;
+bool RandomEnchantAnnounce = true;
+bool Crafted = 1;
+bool Looted = 1;
+bool QuestReward = 1;
+uint32 HighQuality = 5;
+uint32 LowQuality = 1;
+uint32 Chance1 = 60;
+uint32 Chance2 = 65;
+uint32 Chance3 = 70;
+
+class REConfig : public WorldScript
+{
 public:
+	REConfig() : WorldScript("REConfig") { }
+
+	void OnBeforeConfigLoad(bool reload) override
+	{
+		if (!reload) {
+			std::string conf_path = _CONF_DIR;
+			std::string cfg_file = conf_path + "/mod_randomenchants.conf";
+
+			std::string cfg_def_file = cfg_file + ".dist";
+			sConfigMgr->LoadMore(cfg_def_file.c_str());
+			sConfigMgr->LoadMore(cfg_file.c_str());
 
-    RandomEnchantsPlayer() : PlayerScript("RandomEnchantsPlayer") { }
+			// Load Configuration Settings
+			SetInitialWorldSettings();
+		}
+	}
+
+	// Load Configuration Settings
+	void SetInitialWorldSettings()
+	{
+		RandomEnchantEnabled = sConfigMgr->GetBoolDefault("RandomEnchants.Enabled", true);
+		RandomEnchantAnnounce = sConfigMgr->GetBoolDefault("RandomEnchants.Announce", true);
+		Crafted = sConfigMgr->GetBoolDefault("RandomEnchants.OnCreate", 1);
+		Looted = sConfigMgr->GetBoolDefault("RandomEnchants.OnLoot", 1);
+		QuestReward = sConfigMgr->GetBoolDefault("RandomEnchants.OnQuestReward", 1);
+		HighQuality = sConfigMgr->GetIntDefault("RandomEnchants.HighQuality", 1);
+		LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+		Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
```

Comment:
I was suggesting this
```suggestion
		Chance1 = (sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60)) ? sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60) : 60
```

then i realized the "60" is already there and anyway see my comments above (aka default values should stay in the conf for clarity and no duplicate code)